File: C++/herencia.cpp
#include <stdio.h>
#include <iostream>

using namespace std;


class Programador{
        
    protected:
        string nombre;
        int edad;
        string skill = "Programo";
    public:
        Programador(string _nombre, int _edad){
            nombre = _nombre;
            edad = _edad;
        }
    
    string getName(){
        
        return nombre;
    }

    void setName(string n){
        nombre = n;
    }

    void saludar(string n){

        cout << "Hola " << n << " Me llamo " << nombre << endl;

    }

    void SaySkill(){
        cout << "Hola yo " << skill;
    }

        
};


class ProgramadorJavascript : public Programador{
    private:
        string skill = "programo en javascript";
    public:
    ProgramadorJavascript(string _nomb, int _eda) : Programador(_nomb,  _eda){
        nombre =_nomb;
        edad =_eda;
    }

    
};

int main(int argc, char const *argv[])
{
    
    Programador kevin("kevin",17);
    ProgramadorJavascript aron("Aron", 18);
    aron.saludar(kevin.getName());
    kevin.saludar(aron.getName());
    kevin.SaySkill();
    aron.SaySkill();
    return 0;
}

File: Python/herencia.py
class Programador:
    def __init__(self,nombre, edad, habilidad = "programar"):
        self.nombre = nombre
        self.edad = edad 
        self.habilidad = habilidad
    
    def saludar(self):
        print(f'Hola me llamo {self.nombre} y se {self.habilidad}')
    
    def saludar_programador(self,nombre):
        print(f'Hola {nombre.nombre}, me llamo {self.nombre}')


class ProgramadorJavascript(Programador):
    def __init__ (self, nombre, edad, habilidad = "javascript"):
        super().__init__(nombre, edad, habilidad)


if __name__ == "__main__":
    kevin = ProgramadorJavascript("Kevin", 18)
    kevin.saludar()

    juan = Programador("Juan", 20)
    kevin.saludar_programador(juan)